src/seismometer/_api.py

```diff
@@ -176,86 +183,126 @@ def fairness_audit(metric_list: Optional[list[str]] = None, fairness_threshold=1
+    score_column: str,
+    score_threshold: float,
+    per_context: bool = False,
+    metric_list: Optional[list[str]] = None,
+    fairness_threshold: float = 1.25,
+) -> HTML | IFrame:
+    sg = Seismogram()
+    path = "aequitas_{cohorts}_with_{target}_and_{score}_gt_{threshold}_metrics_{metrics}_ratio_{ratio}".format(
+        cohorts="_".join(cohort_columns),
+        target=target_column,
+        score=score_column,
+        threshold=score_threshold,
+        metrics="_".join(metric_list),
+        ratio=fairness_threshold,
+    )
+    if per_context:
+        path += "_grouped"
+    fairness_path = sg.config.output_dir / (slugify(path) + ".html")
+    height = 100 + 100 * len(metric_list)
 
+    if fairness_path.exists():
+        return load_as_iframe(fairness_path, height=height)
 
-@export
-@display_cached_widget
-def cohort_selector():
-    """
-    Display an IPyWidget selector to control cohorts used in visualizations.
-    """
-    from seismometer.controls.selection import DisjointSelectionListsWidget
+    target = pdh.event_value(target_column)
+    data = (
+        pdh.event_score(
+            sg.data(),
+            sg.entity_keys,
+            score=sg.output,
+            ref_event=sg.predict_time,
+            aggregation_method=sg.event_aggregation_method(sg.target),
+        )
+        if per_context
+        else sg.data()
+    )
 
-    sg = Seismogram()
-    options = sg.available_cohort_groups
-    widget = DisjointSelectionListsWidget(
-        options=options, value=sg.selected_cohort, title="Select Subgroups of Interest", select_all=True
+    data = data[[target, score_column] + cohort_columns]
+    if len(data.index) < sg.censor_threshold:
+        return template.render_censored_plot_message(sg.censor_threshold)
+    fairness_audit_to_html(
+        fairness_path, data, cohort_columns, score_column, target, score_threshold, metric_list, fairness_threshold
     )
+    return load_as_iframe(fairness_path, height=height)
 
-    def on_widget_value_changed(*args):
-        sg.selected_cohort = widget.value
 
-    widget.observe(on_widget_value_changed, "value")
+# endregion
 
-    # get intial value
-    on_widget_value_changed()
-    return widget
+# region notebook IPWidgets # TODO - move to .controls submodule
 
 
 @export
-@display_cached_widget
 def cohort_list():
     """
     Displays an exhaustive list of available cohorts for analysis.
     """
     sg = Seismogram()
-    from ipywidgets import HBox, Output
+    from ipywidgets import Output, VBox
 
     from .controls.selection import MultiSelectionListWidget
-    from .data.filter import filter_rule_from_cohort_dictionary
 
     options = sg.available_cohort_groups
 
-    comparison_selections = MultiSelectionListWidget(
-        options, ghost_text="Select cohort groups", title="Available Cohorts"
-    )
+    comparison_selections = MultiSelectionListWidget(options, title="Cohort")
     output = Output()
 
     def on_widget_value_changed(*args):
```

Comment:
*args isn't being used - is this still a necessary parameter?

Reply:
yes, because the caller still sends us args, even if we ignore them.